File: pyg_cart_interface/src/main/java/com/pyg/cart/service/CartService.java
package com.pyg.cart.service;

import com.pyg.group.Cart;

import java.util.List;

public interface CartService {
    public List<Cart> addGoodsToCartList(List<Cart> cartList, Long itemId, Integer num );
    public List<Cart> findCartListFromRedis(String username);
    public void saveCartListToRedis(String username,List<Cart> cartList);
    public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
}

File: pyg_cart_web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){


    //查询购物车列表
    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }

    //添加商品到购物车列表
    this.addGoodsToCartList=function (itemId,num) {
        return $http.get('../cart/addGoodsToCartList.do?itemId='+itemId+'&num=' + num);
    }
    this.findAddressList=function(){
        return $http.get('../address/findListByLoginUser.do');
    }
    this.submitOrder=function(order){
        return $http.post('../order/add.do',order);
    }
});